.github/workflows/release.yml

```diff
@@ -0,0 +1,71 @@
+# Workflow to build release artifacts
+
+name: release artifacts
+
+on:
+  release:
+    types: [ created ]
+
+jobs:
+  build-and-test:
+    strategy:
+      fail-fast: false
+
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Setup Java 8
+      uses: actions/setup-java@v1.4.3
+      with:
+        java-version: '8'
+        java-package: jdk
+        architecture: x64
+    - name: set JDK 6,7,8 environment variables for kotlin compiler
+      env:
+        ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+      run: echo ::set-env name=JDK_16::$(echo $JAVA_HOME) && echo ::set-env name=JDK_17::$(echo $JAVA_HOME) && echo ::set-env name=JDK_18::$(echo $JAVA_HOME)
+    - name: Setup Java 9
+      uses: actions/setup-java@v1.4.3
+      with:
+        java-version: '9'
+        java-package: jdk
+        architecture: x64
+    - name: set JDK_9 environment variable for kotlin compiler
+      env:
+        ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+      run: echo ::set-env name=JDK_9::$(echo $JAVA_HOME)
+
+    # Checkout
+    - uses: actions/checkout@v2
+
+    # Build cache
+    - name: Cache Gradle Cache
+      uses: actions/cache@v2
+      with:
+        path: ~/.gradle/caches
+        key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle.kts') }}-${{ hashFiles('**/gradle.properties') }}
+        # An ordered list of keys to use for restoring the cache if no cache hit occurred for key
+        restore-keys: |
+          ${{ runner.os }}-gradle-
+    - name: Cache gradle wrapper
+      uses: actions/cache@v2
+      with:
+        path: ~/.gradle/wrapper
+        key: ${{ runner.os }}-gradle-wrapper-${{ hashFiles('gradle/wrapper/gradle-wrapper.properties') }}
+    # Build KSP artifacts
+    - name: build
+      run: |
+        REF=${{ github.ref }} # refs/tags/$KSP_VERSION
+        ./gradlew --info -PkspVersion=${REF:10} publishAllPublicationToTestRepository
```

Comment:
it might be better to use -PoutRepo in case we decide to make other changes in test repo publishing (e.g. one day we might want to put KSP to strict mode etc for integration tests but not for release)

Reply:
Done.